src/RadixDlt.NetworkGateway.Abstractions/Configuration/Extensions.cs

```diff
@@ -64,11 +64,45 @@
 
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Options;
+using RadixDlt.NetworkGateway.Abstractions.Exceptions;
+using RadixDlt.NetworkGateway.Abstractions.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace RadixDlt.NetworkGateway.Abstractions.Configuration;
 
 public static class Extensions
 {
+    public static List<CoreApiNode> GetEnabledNodes(this ICollection<CoreApiNode> nodes)
+    {
+        return nodes.Where(n => n.Enabled).ToList();
```

Comment:
any reason we're not throwing here if 0 nodes are enabled? and shouldn't we have separate class responsible for node configuration validation?

Reply:
I've decided not to change the current application logic where 0 enabled nodes are fine for some areas (e.g. healthchecks), even if this unavoidably leads to failure later on. Actually you're right. I've extend our existing configuration validation to ensure at least one Node is enabled.